test/e2e/offset.test.ts

```diff
@@ -204,23 +204,34 @@ describe("offset", () => {
   })
 
   describe("query", () => {
-    it("the consumer is able to track the offset of the stream through queryOffset method", async () => {
-      let offset: bigint = 0n
+    it("the consumer is able to track the offset and start from the stored offset", async () => {
+      const resumedMessages: Message[] = []
+      const publisher = await client.declarePublisher({ stream: testStreamName })
+      await publisher.send(Buffer.from("hello"))
+      await publisher.send(Buffer.from("marker"))
+      await publisher.send(Buffer.from("hello"))
+      await publisher.send(Buffer.from("world"))
+
       const consumer = await client.declareConsumer(
-        { stream: testStreamName, offset: Offset.next(), consumerRef: "my_consumer" },
+        { stream: testStreamName, offset: Offset.first(), consumerRef: "my_consumer" },
         async (message: Message) => {
-          await consumer.storeOffset(message.offset!)
-          offset = message.offset!
+          if (message.content.toString() === "marker") {
+            await consumer.storeOffset(message.offset!)
+          }
+        }
+      )
+      await wait(3000)
```

Comment:
Instead of a fix wait that is always to short or too long, I suggest to find a trigger or for example set a flag that trigger the other behaviour